lib/Test/Nginx/Util.pm

```diff
@@ -243,7 +243,30 @@ our $ForceRestartOnTest     = (defined $ENV{TEST_NGINX_FORCE_RESTART_ON_TEST})
 
 if ($Randomize) {
     srand $$;
-    $ServerPort = int(rand 60000) + 1025;
+
+    for (my $i = 0; $i <= 1000; $i++) {
+        $ServerPort = int(rand 60000) + 1025;
+
+        my $sock = IO::Socket::INET->new(
+            PeerAddr => $ServerAddr,
+            PeerPort => $ServerPort,
```

Comment:
Use of `PeerAddr` and `PeerPort` is wrong. They are *remote* addresses to connect to. We need a listening socket instead of a client socket.